lab-said/frontend/src/component/child-form/index.js

```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+import * as utils from '../../lib/util';
+
+class ChildForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = props.child ? props.child : {name: ''};
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  componentWillReceiveProps(props) {
+    if(props.child) this.setState(props.child);
+  }
+
+  handleChange(e) {
+    this.setState({name: e.target.value});
+  }
+
+  handleSubmit(e) {
+    e.preventDefault();
+    let {onComplete} = this.props;
+    let result = onComplete(this.state);
+    if (result instanceof Promise) {
+      result
+      .then(() => this.setState({ error: null }))
+      .catch(error => {
+        util.log('ListForm Error:', error);
```

Comment:
This line would break if an error was thrown. Right now html5 is handling enough validations that this block wont likely be reached. However, the simple util. vs utils. typo will break if it did reach the error code block.